gstd/src/msg/async.rs

```diff
@@ -51,15 +53,24 @@ where
             "Somebody created a future with the MessageId that never ended in static replies!"
         ),
         ReplyPoll::Pending => Poll::Pending,
-        ReplyPoll::Some((actual_reply, reply_code)) => {
+        ReplyPoll::Some((payload, reply_code)) => {
             // Remove lock after waking.
             async_runtime::locks().remove(msg_id, waiting_reply_to);
 
-            if !reply_code.is_success() {
-                return Poll::Ready(Err(Error::ReplyCode(reply_code)));
-            }
+            match reply_code {
+                ReplyCode::Success(_) => Poll::Ready(f(payload)),
+                ReplyCode::Error(reason) => {
+                    let err = String::from_utf8(payload)
```

Comment:
at this level, you should not consider it a string

Better wrap it in different error elsewhere

For example, put payload it in wrapped struct and add  `fn as_string() -> Option<String>` for it

Reply:
But payloads of error replies are always strings atm 
https://github.com/gear-tech/gear/blob/5922c049a78ac14ce37f839fa30c0647080ceef9/core-processor/src/processing.rs#L264-L270

Reply:
Or you just want to avoid extra calculations in case of somebody ignore err?

Reply:
also this, but better not to rely that something is string

protocol itself does not dictate strings in the payload

Reply:
What the problem with debug? You can convert it in display / debug impl...